### ISSUE TITLE
Issues (potentially with) 'waitForSync: true'

### ISSUE THREAD

**Issue by prodrammer** [arangodb/arangodb-java-driver-async#11]
I'm running arangodb `3.3.3` via docker container. There is a single instance (no clustering).

I'm doing essentially the following:

- `Statement 1`: UPSERT...INSERT....REPLACE....OPTIONS { waitForSync: true }
- `Statement 2`: // Use object inserted in `Statement 1`

Initially, I wasn't using the `waitForSync: true` option and was getting errors stating the object didn't exist when `Statement 2` executed. I thought I had solved it by adding `waitForSync: true`, but the error seemed to reappear. I patched it with a very hackish and arbitrary delay between the two statements. So far, "it's working on my machine"™ but this is obviously something that concerns me.

- Are you aware of any issues with `waitForSync` and this java async driver?
- How does `waitForSync` work with the driver?
- How can I test to ensure that this issue will not appear once I remove the delay?

If I could introduce an artificial delay in the `INSERT` statement, I would be able to test that the `waitForSync` option is observed. For example, imagine executing something like `INSERT...OPTIONS { delay: 5000, waitForSync: false }`. This would return immediately (the `waitForSync` option is not `true`), but would delay execution of the statement. Since the call returns immediately, we can assert that the object does not yet exist via a second query that executes before the delay expires. We can also run `INSERT...OPTIONS { delay: 5000, waitForSync: true }`. Since we enabled `waitForSync` the call doesn't return until the record is inserted. We can assert that the object does exist via a subsequent query.

If I can test and get positive results that `waitForSync` is behaving _and I still experience the same issue_, then at least we know it's not the `waitForSync` setting, and perhaps we can explore other options together.

Appreciate any guidance you can offer!

Thanks!

**Comment by Simran-B** [comment]
Regarding an artificial delay, there is `SLEEP()` in AQL:

```
UPSERT...INSERT....REPLACE....OPTIONS...
LET x = SLEEP(5)
RETURN NEW
```

You can place SLEEP at the top as well.

**Comment by mvollmary** [comment]
@ryanhaney If `waitForSync` is set to true then the data is synchronized to disk before returning from a document create, update, replace or removal operation. Independent of `waitForSync` you should be able to work with the object you created in statement 1. Can you share your source code with the two performed statements?

**Comment by prodrammer** [comment]
TL;DR; - I removed my "hackish and arbitrary" delays and I can't reproduce the error. I am worried that I haven't properly squashed this issue. I'd like to be able to prove that this is not at all related to `waitForSync`. At this point, _I'm not even sure it's related to arango at all_.

How would one go about testing `waitForSync` to ensure it is working as advertised? What is the use case? How can I produce a situation that would require `waitForSync: true` to operate, assert it fails using `waitForSync: false`, then set `waitForSync` to `true` and assert it passes? Adding these tests to our testing suite would give me peace of mind knowing that arango is working as expected.

Thanks!

**Comment by prodrammer** [comment]
I'm going to close this issue until I can prove it needs to be redressed.

I appreciate your patience and assistance on this.
